analyzers/src/SonarAnalyzer.CSharp/Rules/CheckArgumentException.cs

```diff
@@ -153,11 +153,12 @@ static IEnumerable<string> AccessorIdentifierNames(AccessorDeclarationSyntax acc
             }
 
             static IEnumerable<string> ParentParameterList(SyntaxNode node) =>
-                node?.Ancestors().OfType<BaseTypeDeclarationSyntax>().FirstOrDefault() is { } baseTypeAncestor
-                && RecordDeclarationSyntaxWrapper.IsInstance(baseTypeAncestor)
-                && ((RecordDeclarationSyntaxWrapper)baseTypeAncestor).ParameterList is { } recordParameterList
-                    ? IdentifierNames(recordParameterList)
+                node?.Ancestors().OfType<TypeDeclarationSyntax>().FirstOrDefault() is { } typeDeclaration
+                    ? GetIdentifierNames(((TypeDeclarationSyntaxWrapper)typeDeclaration).ParameterList)
```

Comment:
Should use the extension method that is implemented here instead:
https://github.com/SonarSource/sonar-dotnet/pull/8209/files#diff-0aed6645a137b426dc5eb96dcf45438091d2b6a5dba073c7c688407106ab9f35R54-R61